git_machete/client.py

```diff
@@ -1530,21 +1535,33 @@ def handle_untracked_branch(self, new_remote: str, branch: str, is_called_from_t
         other_remote_choice = "o[ther-remote]" if can_pick_other_remote else ""
         remote_branch = f"{new_remote}/{branch}"
         if not self.__git.get_commit_sha_by_revision(remote_branch, prefix="refs/remotes/"):
-            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'q', other_remote_choice))
+            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'Q', other_remote_choice))
             ask_message = f"Push untracked branch {bold(branch)} to {bold(new_remote)}?" + choices
             ask_opt_yes_message = f"Pushing untracked branch {bold(branch)} to {bold(new_remote)}..."
             ans = self.ask_if(ask_message, ask_opt_yes_message,
                               override_answer=None if self.__git.cli_opts.opt_push_untracked else "N")
-            if ans in ('y', 'yes', 'yq'):
-                self.__git.push(new_remote, branch)
-                if ans == 'yq':
-                    raise StopTraversal
-                self.flush_caches()
-            elif can_pick_other_remote and ans in ('o', 'other'):
-                self.__pick_remote(branch, is_called_from_traverse)
-            elif ans in ('q', 'quit'):
-                raise StopTraversal
-            return
+            if is_called_from_traverse:
+                if ans in ('y', 'yes', 'yq'):
+                    self.__git.push(new_remote, branch)
+                    if ans == 'yq':
+                        raise StopInteraction
+                    self.flush_caches()
+                elif can_pick_other_remote and ans in ('o', 'other'):
+                    self.__pick_remote(branch, is_called_from_traverse)
+                elif ans in ('q', 'quit'):
+                    raise StopInteraction
+                return
+            else:
+                if ans in ('y', 'yes'):
+                    self.__git.push(new_remote, branch)
+                    self.flush_caches()
+                elif can_pick_other_remote and ans in ('o', 'other'):
+                    self.__pick_remote(branch, is_called_from_traverse)
+                elif ans in ('q', 'quit'):
+                    raise StopInteraction
+                else:
+                    raise MacheteException(f'Cannot create pull request from untracked branch `{branch}`')
+                return
```

Comment:
Nit: this `return` will never be reached

Reply:
will be, when `y` is selected. Maybe let's move it couple lines up.

Reply:
Right! didn't notice